### ISSUE TITLE
how do you differentiate if user tapped back button or ok (tick) button

### ISSUE THREAD

**Issue by veenatic** [Sh1d0w/multi_image_picker#185]
I am trying to cover scenario when user has already picked some images, but he goes to the gallery again to pick something, he decides to not pick any, clicks back button thinking that what he previously selected, remains. What i am seeing that when user clicks back button, the pickImages function returns success with 0 assets. This behavior is same as user not selecting anything and clicking the ok (tick) button. It returns 0 assets too. Therefore, i cant be sure if i should keep the previously selected images or discard them.

**Comment by Sh1d0w** [comment]
This does not concerns the plugin, it is up for your implementation. If you want to make funcationality where the user can remove images just add remove icons or something on the images or a clear all button rather than relying on the result from the image picker.

**Comment by veenatic** [comment]
well, i am not talking about user removing items or clearing all. its about the fact that the plugin does not differentiate in any way if user clicked the back button or the tick button. in both cases it returns success and returns the selected assets only if tick button was clicked. if the user did not select any assets, clicking back button or tick button will return the same 0 assets.

Its not a big issue but from the user experience perspective, it is not good. the scenario is similar to a situation where user gets a dialog with ok and cancel button, and you dont know if user clicked ok or cancel..

**Comment by Sh1d0w** [comment]
I understand you now. Yes it will be possible to implement such thing. Since the plugin returns Future with Assets array, I think the only way to implement this is if he pressed cancel on iOS or back on Android an exception with name similar to `NoImagesSelected` to be thrown, so you will be able to handle it like this:

```
try {
      resultList = await MultiImagePicker.pickImages(
          maxImages: 300,
          enableCamera: true
          ));
    } on NoImagesSelected catch (e) {
      // Do some stuff here if no images are selected
    } on PlatformException catch (e) {
      error = e.message;
    }
```

What do you think?

**Comment by veenatic** [comment]
Precisely. Many times you would cache the previously selected images so that during widget rebuild, it does not loose them. So, in case user pressed back button, the previously selected images can be reused (and not discarded because back button returned 0 images).

**Comment by Sh1d0w** [comment]
Hi, this has been implemented in version 4.4.0

`pickImages` now throws NoImagesSelectedException, PermissionDeniedException and PermissionPermanentlyDeniedExeption in addition to PlatformException. You can implement different logic for handling each case separately now.
